File: 1730-special-array-with-x-elements-greater-than-or-equal-x/special-array-with-x-elements-greater-than-or-equal-x.py
class Solution(object):
    def specialArray(self, nums):
        n = len(nums)
        prev = -1
        nums.sort()

        for i in range(n):
            if(n - i <= nums[i] and n - i > prev):
                return n - i
            prev = nums[i]
        
        return -1

File: 1730-special-array-with-x-elements-greater-than-or-equal-x/special-array-with-x-elements-greater-than-or-equal-x.c
#include <stdio.h>
#include <stdlib.h>

int cmpfunc (const void * a, const void * b) {
   return ( *(int*)a - *(int*)b );
}

int specialArray(int* nums, int numsSize) {
    qsort(nums, numsSize, sizeof(int), cmpfunc);
    int prev = -1;
    for (int i = 0; i < numsSize; i++) {
        if (numsSize - i <= nums[i] && numsSize - i > prev) {
            return numsSize - i;
        }
        prev = nums[i];
    }
    return -1;
}

